recipes/libcusolver/meta.yaml

```diff
@@ -0,0 +1,181 @@
+{% set name = "libcusolver" %}
+{% set version = "11.4.2.57" %}
+{% set cuda_version = "12.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: c9460db500d0c7d14debb2c8e0d88fc26a39197363f1b000ccf760866f636f12  # [linux64]
+  sha256: bf3c9a090652d011aafe2190c6f26fe108194cf8512d3c571e406054740168bb  # [ppc64le]
+  sha256: 0ca46b73520657f96f921711c05af7053bd398328d510aecddfff2521c000e91  # [aarch64]
+  sha256: c09836df1f43a3496f2b61ad8827af662ace298508869564a40c30d909ba550f  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: libcusolver
+    files:
+      - lib/libcusolver*.so.*                            # [linux]
+      - targets/{{ target_name }}/lib/libcusolver*.so.*  # [linux]
+      - Library\bin\cusolver*.dll                        # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+        - libcublas
+        - libnvjitlink
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ pin_compatible("libcublas", max_pin="x") }}
```

Comment:
The `cuda-version` pin here will force the `libcublas` minor versions anyway. Should we pin to the minor version to make thinks more clear?

Reply:
Yes, for now this doesn't matter since we are guarded by cuda-version. I slightly prefer to have the pinning on cublas/nvjitlink relaxed, in case in the future we want to allow certain degree of mix-n-match. But I don't have strong opinion here.

Reply:
If the library only requires the major version as opposed to newer than the matching minor version then I'm generally a -1 to having `cuda-version` be a run requirement as opposed to run_constrained.

Reply:
Based on the summary in https://github.com/conda-forge/cuda-version-feedstock/issues/1#issuecomment-1462998398 (and the discussion around it) I think we agreed that
- `cuda-version` needs to be a run requirement so that 1. we can use it as an identifier to declare which CUDA Version a package supports, and 2. later we can request the exact components for a certain CTK version can be requested via `cuda-toolkit` & co.
- In theory CTK components from within the same major version could be mix-n-matched (ex: cuSOLVER v11 depends on cuBLAS v12), but for now we're using `cuda-version` as a safe guard to disallow this, and will revisit this decision in the future. 

I'd prefer to stick to the above conclusion for now, and raise an issue in the created feedstocks to track this discussion. I don't think it's a blocker given that mix-n-match is still disallowed. Otherwise, this discussion should be brought back to cuda-version-feedstock, not here.

Reply:
Then I think we should pin to the minor version of `libcublas` here to reduce the ambiguity of what our dependency constraints are at this time.

Reply:
See d6f084e.

Reply:
Opened two issues in the feedstocks to track this discussion:
- https://github.com/conda-forge/libcusolver-feedstock/issues/5
- https://github.com/conda-forge/libcusparse-feedstock/issues/4